### ISSUE TITLE
add Python 2.6 support

### ISSUE THREAD

**Issue by jonisb** [jonisb/PythonColorer#21]
Python 2.6 syntax will be the base syntax, I'll list differences to other versions here:

Difference between 2.6 and 2.7.x:

- [ ] set literal syntax missing: [Other Language Changes](https://docs.python.org/2/whatsnew/2.7.html#other-language-changes)
- [ ] with statement changed: [Other Language Changes](https://docs.python.org/2/whatsnew/2.7.html#other-language-changes)

Difference between 2.6 and 3.4:

- [ ] Default source encoding changed from ASCII to UTF-8: [The default source encoding is now UTF-8](https://www.python.org/dev/peps/pep-3120)
- [ ] Non-ASCII letters are now allowed in identifiers.: [PEP 3131: Non-ASCII letters are now allowed in identifiers.](https://www.python.org/dev/peps/pep-3131)
- [ ] str changed to bytes and Unicode changed to str: [Text Vs. Data Instead Of Unicode Vs. 8-bit](https://docs.python.org/3/whatsnew/3.0.html#text-vs-data-instead-of-unicode-vs-8-bit)
- [ ] print is a function: [Print Is A Function](https://docs.python.org/3/whatsnew/3.0.html#print-is-a-function)
- [ ] new raise statement syntax: raise [expr [from expr]]: [Changed Syntax](https://docs.python.org/3/whatsnew/3.0.html#changed-syntax)
- [ ] True, False, and None are reserved words. (2.6 partially enforced the restrictions on None already.): [Changed Syntax](https://docs.python.org/3/whatsnew/3.0.html#changed-syntax)
- [ ] Change from except exc, var to except exc as var.: [Changed Syntax](https://docs.python.org/3/whatsnew/3.0.html#changed-syntax)
- [ ] List comprehensions no longer support the syntactic form [... for var in item1, item2, ...].: [Changed Syntax](https://docs.python.org/3/whatsnew/3.0.html#changed-syntax)
- [ ] New Metaclass Syntax.: [Changed Syntax](https://docs.python.org/3/whatsnew/3.0.html#changed-syntax)
- [ ] The ellipsis (...) can be used as an atomic expression anywhere. (Previously it was only allowed in slices.) Also, it must now be spelled as .... (Previously it could also be spelled as . . ., by a mere accident of the grammar.): [Changed Syntax](https://docs.python.org/3/whatsnew/3.0.html#changed-syntax)
- [ ] long renamed int no longer uses "L" affix : [Integers](https://docs.python.org/3/whatsnew/3.0.html#integers)
- [ ] Octal literals are no longer of the form 0720; use 0o720 instead. : [Integers](https://docs.python.org/3/whatsnew/3.0.html#integers)
- [ ] Function argument and return value annotations: [PEP 3107: Function argument and return value annotations](https://www.python.org/dev/peps/pep-3107)
- [ ] Keyword-only arguments: [PEP 3102: Keyword-only arguments](https://www.python.org/dev/peps/pep-3102)
- [ ] Keyword arguments are allowed after the list of base classes in a class definition: [New Syntax](https://docs.python.org/3/whatsnew/3.0.html#new-syntax)
- [ ] nonlocal statement.: [PEP 3104: nonlocal statement.](https://www.python.org/dev/peps/pep-3104)
- [ ] Extended Iterable Unpacking: [PEP 3132: Extended Iterable Unpacking](https://www.python.org/dev/peps/pep-3132)
- [ ] Removed backticks (use repr() instead).: [Removed Syntax](https://docs.python.org/3/whatsnew/3.0.html#removed-syntax)
- [ ] Tuple parameter unpacking removed.: [PEP 3113: Tuple parameter unpacking removed.](https://www.python.org/dev/peps/pep-3113)
- [ ] Removed keyword: exec() is no longer a keyword: [Removed Syntax](https://docs.python.org/3/whatsnew/3.0.html#removed-syntax)
- [ ] The from module import * syntax is only allowed at the module level, no longer inside functions.: [Removed Syntax](https://docs.python.org/3/whatsnew/3.0.html#removed-syntax)
- [ ] The only acceptable syntax for relative imports is from .[module] import name. All import forms not starting with . are interpreted as absolute imports.: [Removed Syntax](https://docs.python.org/3/whatsnew/3.0.html#removed-syntax)

Auto generated syntax:

- [ ] BuiltinFunctions
- [ ] BuiltinExceptions
- [ ] Special names

**Comment by hugovk** [comment]
Unless you _really_ need to use Python 2.6, I probably wouldn't bother:

* Unsupported since [2013-10-29](https://en.wikipedia.org/wiki/CPython#Version_history), no longer receiving security updates
* https://snarky.ca/stop-using-python-2-6/
* http://www.curiousefficiency.org/posts/2015/04/stop-supporting-python26.html
* http://www.python3statement.org
* Current pip 9 deprecates Python 2.6 support, pip 10 won't support it (https://github.com/pypa/pip/issues/3955)
* Not much PyPI traffic (June 2016) https://github.com/pypa/pip/issues/3796

**Comment by jonisb** [comment]
Yes I know it's old and obsolete but I need it for a project and the syntax is not too different from 2.7 so is not too much extra work.

But thanks for the heads-up

jonib
